snmp/tests/test_profiles.py

```diff
@@ -0,0 +1,1024 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import os
+
+import pytest
+
+from datadog_checks import snmp
+from datadog_checks.snmp import SnmpCheck
+
+from . import common
+from .utils import mock_profiles_root
+
+pytestmark = pytest.mark.usefixtures("dd_environment")
+
+
+@pytest.fixture(autouse=True)
+def host_profiles_root():
+    # By default, we resolve profiles relative to the `snmp.d` directory.
+    # But this directory is only created by the Agent when the integration is installed, so we have
+    # to replace it with the path within the Python package.
+    package_profiles_root = os.path.join(os.path.dirname(snmp.__file__), 'data', 'profiles')
+
+    with mock_profiles_root(package_profiles_root):
+        yield
+
+
+def run_profile_check(profile, community, profile_name, set_profile=False):
+    """Run a single check with the given `profile`, with the provided `community`.
+
+    If `set_profile` is True, we don't rely on `sysObjectID` matching and pass the profile explicitly.
+    """
+    instance = common.generate_instance_config([])
+
+    instance['community_string'] = community
+    instance['enforce_mib_constraints'] = False
+    if set_profile:
+        instance['profile'] = profile_name
+
+    init_config = {'profiles': {profile_name: {'definition_file': profile}}}
```

Comment:
I feel like a better testing pattern would be to have all profiles defined here in the init_config, and just test what the snmp check collects. That could help with potential profiles collisions for example.

I understand this would probably submit more metrics than what we assert now so it's probably out of the scope of this PR, but I'm curious what your thoughts are on this.

Reply:
It shouldn't produce more metrics, as we only match one profiles.

We also have a potential feature request to include all by defaults, so that aligns with that. But yeah if you don't mind we'll do it late.

Reply:
But we do have `extends` now that are not taken into account at the moment in tests, right?

Reply:
No, extends are taken into account in tests currently.